syntax = "proto3";

package aelf;

message Transaction {
    // The address of the sender of the transaction.
    Address from = 1; // Tag = 0x0a
    // The address of the contract when calling a contract.
    Address to = 2; // Tag = 0x12
    // The height of the referenced block hash.
    int64 ref_block_number = 3; // Tag = 0x18
    // The first four bytes of the referenced block hash.
    bytes ref_block_prefix = 4; // Tag = 0x22
    // The name of a method in the smart contract at the To address.
    string method_name = 5; // Tag = 0x2a
    // The parameters to pass to the smart contract method.
    bytes params = 6; // Tag = 0x32
    // When signing a transaction itâ€™s actually a subset of the fields: from/to and the target method as well as 
    // the parameter that were given. It also contains the reference block number and prefix.
    bytes signature = 10000; // Tag = 0x82f104
}

message Address
{
    bytes value = 1;
}

message TransferInput {
    // The receiver of the token.
    Address to = 1; // Tag = 0x0a
    // The token symbol to transfer.
    string symbol = 2; // Tag = 0x12
    // The amount to to transfer.
    int64 amount = 3; // Tag = 0x18
    // The memo.
    string memo = 4; // Tag = 0x22
}
